src/dependency-manager/src/manager.ts

```diff
@@ -127,13 +128,16 @@ export default abstract class DependencyManager {
     }
 
     // Load component dependencies
-    if (load_dependencies) {
-      for (const [dep_key, dep_value] of Object.entries(component.getDependencies())) {
-        const dep_name = dep_value.includes(':') ? `${dep_key}:latest` : `${dep_key}:${dep_value}`;
-        const dep_extends = dep_value.includes(':') ? dep_value : `${dep_key}:${dep_value}`;
-        const dep_component = ComponentConfigBuilder.buildFromJSON({ extends: dep_extends, name: dep_name });
-        await this.loadComponent(graph, dep_component, component_map);
-      }
+    for (const [dep_key, dep_value] of Object.entries(component.getDependencies())) {
+      const dep_name = dep_value.includes(':') ? `${dep_key}:latest` : `${dep_key}:${dep_value}`;
+      const dep_extends = dep_value.includes(':') ? dep_value : `${dep_key}:${dep_value}`;
+      const dep_component = ComponentConfigBuilder.buildFromJSON({ extends: dep_extends, name: dep_name });
+
+        if (dep_component.getRef() in component_map) { // Reject circular dependencies
+          if (loaded_by_env.includes(dep_component.getRef())) continue;
```

Comment:
Instead of `loaded_by_env` can you check the dep_component.getDependencies() directly since its already loaded? I think we only one to reject direct dependencies not implicit.

Add a test to make sure we don't reject the following: 
i.e. 
Component A depends on Component B and C
Component B depends on C